### ISSUE TITLE
[Accessibility]: extension actions on search page should be buttons

### ISSUE THREAD

**Issue by tjkandala** [sourcegraph/sourcegraph#36752]
### Audit type

General/Other

### User journey audit issue

#34054

### Problem description

On the search page with the `search-export` extension enabled, the action contributed by `search-export` is rendered as  `<a>` and not `<button>`, despite not causing page navigation on click.

### Expected behavior

Search page extension actions should be `<buttons>` and look like the built-in actions.

### Additional details

![Screen Shot 2022-06-07 at 5 15 42 PM](https://user-images.githubusercontent.com/37420160/172484172-64571cd7-2bba-4c33-9478-4ad33bc79d19.png)
 
The styling difference affects usability on mobile too:

![Screen Shot 2022-06-07 at 5 19 39 PM](https://user-images.githubusercontent.com/37420160/172484663-013e9402-45a3-482f-84ac-0981bf23766f.png)

Button styling would increase the size of the touch target.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [x] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [x] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
